<a href="https://colab.research.google.com/github/Chandu0444/Machine_learning/blob/main/PCA_MC(23_10_2025).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
data_set

In [ ]:
import numpy as np
import pandas as pd

In [ ]:
data=pd.DataFrame({'x1':[5,6,4,5,6],'x2':[9,10,6,8,9],'Target':['F','M','F','M','M']})

In [ ]:
data

x1  x2 Target
0   5   9      F
1   6  10      M
2   4   6      F
3   5   8      M
4   6   9      M

In [ ]:
covarinace_val1=data['x1'].cov(data['x2']) #returns (x1,x2) covvariance
covarinace_val1

np.float64(1.15)

In [ ]:
covarinace_val2=data['x2'].cov(data['x1']) #returns (x2,x1) covvariance
covarinace_val2

np.float64(1.15)

In [ ]:
covarinace_val3=data['x1'].cov(data['x1']) #returns (x1,x1) covvariance
covarinace_val3

np.float64(0.7)

In [ ]:
covarinace_val4=data['x2'].cov(data['x2']) #returns (x2,x2) covvariance
covarinace_val4

np.float64(2.3)

In [ ]:
covariance_mat=np.array([[0.7,1.15],[1.15,2.3]])
covariance_mat

array([[0.7 , 1.15],
       [1.15, 2.3 ]])

In [ ]:
eig_val, eig_vec= np.linalg.eig(covariance_mat)

In [ ]:
eig_val

array([0.09910743, 2.90089257])

In [ ]:
eig_vec -0.46310664 -0.88630257

array([[-0.88630257, -0.46310664],
       [ 0.46310664, -0.88630257]])

For 0.09910743 ->  -0.88630257 0.46310664

    2.90089257 -> -0.46310664 -0.88630257

now we choose vector which having hisghest eigen value : 2.90089257

now find the dot product

[5 - 5.2 , 9 - 8.4] -> [-0.2 , 0.6]

....,

....,

....,

[6 - 5.2 ,  9 - 8.4] -> [0.8 , 0.6]


In [ ]:
X1 = np.array([-0.46,-0.88])
X1_transpose = X1.transpose()
X2 = np.array([-0.2,0.6])

In [ ]:
np.dot(X1_transpose,X2)

np.float64(-0.436)

dot products are: -0.436 .,,,

In [ ]:
X = np.array([[5,9],
              [6,10],
              [4,6],
              [5,8],
              [6,9]])

In [ ]:
from sklearn.decomposition import PCA

In [ ]:
pca = PCA(n_components=2)

In [ ]:
pca.fit_transform(X)

array([[ 0.43916021, -0.4551245 ],
       [ 1.78856942, -0.03192856],
       [-2.68285413,  0.04789284],
       [-0.44714235,  0.00798214],
       [ 0.90226685,  0.43117807]])

In [ ]:
from sklearn.datasets import load_iris